js/websocket_app/src/stream/Data_Stream.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from 'react';
+import useWebSocket, { ReadyState } from 'react-use-websocket';
+
+function Data_Stream() {
+    const [displayText, setDisplayText] = useState('');
+    const [websocketUrl, setWebsocketUrl] = useState('ws://localhost:8080/ws');
+    const {
+        sendJsonMessage,
+        lastJsonMessage,
+        readyState,
+    } = useWebSocket(websocketUrl, {
+        onOpen: () => console.log('opened'),
+        shouldReconnect: () => true,
+    });
+
+    const connectionStatus = {
+        [ReadyState.CLOSED]: 'Closed',
+        [ReadyState.CLOSING]: 'Closing',
+        [ReadyState.CONNECTING]: 'Connecting',
+        [ReadyState.OPEN]: 'Open',
+        [ReadyState.UNINSTANTIATED]: 'Uninstantiated',
+    }[readyState];
```

Comment:
[Nitpick / readability] Small readability nitpick here, I would define the object outside of the react function and then use this here as just an indexing of said object. 

 ```suggestion
    const CONNECTION_STATUS = {
        [ReadyState.CLOSED]: 'Closed',
        [ReadyState.CLOSING]: 'Closing',
        [ReadyState.CONNECTING]: 'Connecting',
        [ReadyState.OPEN]: 'Open',
        [ReadyState.UNINSTANTIATED]: 'Uninstantiated',
    } as const;
    
    ...
    
    
    const connectionStatus = CONNECTION_STATUS[readyState]
```

Reply:
Also done! Thank you!